NullPointerException in IntegrationManager.java

Using 3.1.1 

We are getting NPE from this - I guess the only possibility is that integrationManagerHandler is null
```
  private void dispatchEnqueue(IntegrationOperation operation) {
    integrationManagerHandler.sendMessage(integrationManagerHandler //
        .obtainMessage(IntegrationManagerHandler.REQUEST_ENQUEUE_OPERATION, operation));
  }
```

From our internal bug discoverer

```We're observing a randomly reproducible NPE (see log.txt) when running espresso tests.
An instance of com.segment.analytics.IntegrationManager is registered for activity lifecycle 
callbacks and the crash happens when it receives the onActivityCreated event
dispatched by android.app.Application.```

Here is the Stack trace 

```
07-29 10:58:34.867  6234  6234 E GoogleInstr: Dying now...
07-29 10:58:34.867  6234  6234 D AndroidRuntime: Shutting down VM
07-29 10:58:34.867  6234  6234 W dalvikvm: threadid=1: thread exiting with uncaught exception (group=0x41c83ba8)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: FATAL EXCEPTION: main
07-29 10:58:34.867  6234  6234 E AndroidRuntime: Process: com.here.app.maps, PID: 6234
07-29 10:58:34.867  6234  6234 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.here.app.maps/com.here.app.MainActivity}: java.lang.NullPointerException
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2184)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2233)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at android.app.ActivityThread.access$800(ActivityThread.java:135)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1196)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:102)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:136)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:5001)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at java.lang.reflect.Method.invokeNative(Native Method)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Method.java:515)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:785)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:601)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at dalvik.system.NativeStart.main(Native Method)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: Caused by: java.lang.NullPointerException
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at com.segment.analytics.IntegrationManager.dispatchEnqueue(IntegrationManager.java:352)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at com.segment.analytics.IntegrationManager.onActivityCreated(IntegrationManager.java:324)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at android.app.Application.dispatchActivityCreated(Application.java:189)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at android.app.Activity.onCreate(Activity.java:903)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at android.support.v4.app.FragmentActivity.onCreate(FragmentActivity.java:255)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at com.here.components.core.BaseActivity.doOnCreate(BaseActivity.java:315)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at com.here.components.states.StatefulActivity.doOnCreate(StatefulActivity.java:610)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at com.here.mapcanvas.states.MapStateActivity.doOnCreate(MapStateActivity.java:219)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at com.here.app.MainActivity.doOnCreate(MainActivity.java:261)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at com.here.components.core.BaseActivity.onCreate(BaseActivity.java:262)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:5231)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at com.google.android.apps.common.testing.testrunner.GoogleInstrumentation.callActivityOnCreate(GoogleInstrumentation.java:428)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2148)
07-29 10:58:34.867  6234  6234 E AndroidRuntime: 	... 11 more
07-29 10:58:34.877  6192  6192 D AndroidRuntime: Shutting down VM
```

--- a/analytics-core/src/main/java/com/segment/analytics/IntegrationManager.java
+++ b/analytics-core/src/main/java/com/segment/analytics/IntegrationManager.java
@@ -93,8 +93,6 @@ class IntegrationManager implements Application.ActivityLifecycleCallbacks {
     this.projectSettingsCache = projectSettingsCache;
     this.logLevel = analytics.getLogLevel();
 
-    application.registerActivityLifecycleCallbacks(this);
-
     integrationManagerThread =
         new HandlerThread(INTEGRATION_MANAGER_THREAD_NAME, THREAD_PRIORITY_BACKGROUND);
     integrationManagerThread.start();
@@ -117,6 +115,7 @@ class IntegrationManager implements Application.ActivityLifecycleCallbacks {
     } else {
       dispatchFetchSettings();
     }
+    application.registerActivityLifecycleCallbacks(this);
   }
 
   private void loadIntegrations() {

